core/api/src/main/java/org/eclipse/sensinact/core/whiteboard/AbstractDescriptiveReadOnly.java

```diff
@@ -0,0 +1,48 @@
+/*********************************************************************
+* Copyright (c) 2024 Kentyou.
+*
+* This program and the accompanying materials are made
+* available under the terms of the Eclipse Public License 2.0
+* which is available at https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*   Thomas Calmant (Kentyou) - initial implementation
+**********************************************************************/
+package org.eclipse.sensinact.core.whiteboard;
+
+import java.lang.reflect.ParameterizedType;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+import org.eclipse.sensinact.core.twin.TimedValue;
+import org.osgi.util.promise.Promise;
+import org.osgi.util.promise.PromiseFactory;
+
+public abstract class AbstractDescriptiveReadOnly<T> implements WhiteboardGet<T>, WhiteboardResourceDescription<T> {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Class<T> getResourceType() {
+        return (Class<T>) ((ParameterizedType) getClass().getGenericSuperclass()).getActualTypeArguments()[0];
```

Comment:
This is fine for now, but may need to be enhanced to cope with types that don't directly extend this class

Reply:
Agreed